api/src/main/java/org/openmrs/module/fhir/api/condition/ObsConditionStrategy.java

```diff
@@ -33,4 +57,70 @@ public Condition getConditionById(String uuid) {
 	public List<Condition> searchConditionByName(String name) {
 		return null;
 	}
+
+	/**
+	 * Returns the fhir Condition for the given OpenMrs Obs. Mapping is done as follows
+	 * FhirCondition.Identifier: Obs.uuid
+	 * FhirCondition.Patient: Link to FHIR Patient
+	 * FhirCondition.Encounter: Link to FHIR Encounter resource
+	 * FhirCondition.Asserter: Link to FHIR Provider
+	 * FhirCondition.DateAsserted: Date created
+	 * FhirCondition.Code: concept mappings for this concept
+	 * FhirCondition.Note: Obs.comment
+	 *
+	 * @param openMrsObs Openmrs Obs object to be mapped to fhir Condition
+	 * @return
+	 */
+	private Condition generateFHIRConditionForOpenMRSObs(Obs openMrsObs) {
+
+		ca.uhn.fhir.model.dstu2.resource.Condition fhirCondition = new ca.uhn.fhir.model.dstu2.resource.Condition();
+		IdDt id = new IdDt();
+		id.setValue(openMrsObs.getUuid());
+		fhirCondition.setId(id);
+
+		//Set patient reference
+		if (openMrsObs.getPerson().isPatient()) {
+			ResourceReferenceDt patient = FHIRUtils.buildPatientOrPersonResourceReference(openMrsObs.getPerson());
+			fhirCondition.setPatient(patient);
+		}
+		//Set Encounter
+
```

Comment:
Still have to identify how the encounter and asserter is going to be mapped correctly. That's why these comments are there. Once they are done I will remove these comments.